internal/redfishwrapper/firmware.go

```diff
@@ -99,6 +100,14 @@ func (c *Client) FirmwareUpload(ctx context.Context, updateFile *os.File, params
 		)
 	}
 
+	// For X13 the full Task structure is returned in the body.   If we can Unmarshall then we can safely assume
+	// that redfishTask.ID contains the ID.
+	redfishTask := &redfish.Task{}
+	err = json.Unmarshal(response, redfishTask)
```

Comment:
Just a note, if response is valid JSON but isn't a `redfish.Task` I believe you'll get no error and zero value for the `Task`.  It might be worth a sanity check that the task id isn't an empty string here.